tests/test-wibox-shape.lua

```diff
@@ -18,14 +18,103 @@ local wb = wibox {
 wb:geometry(screen[1].geometry)
 wb.visible = true
 
-runner.run_steps({
+local count = 0
+
+local steps = {
     function()
         assert(wb.shape == shape.powerline)
         assert(wb.shape_bounding) -- This is a memory leak! Don't copy!
         if was_drawn then
             return true
         end
+    end,
+    -- Remove the shape and test the input
+    function()
+        wb.shape = nil
+        wb.input_passthrough = false
+        wb:connect_signal("button::press", function()
+            count = count + 1
+        end)
+
+        wb:geometry {
+            x      = 0,
+            y      = 100,
+            width  = 101,
+            height = 101,
+        }
+        wb.border_width = 0
+
+        return true
     end
-})
+}
+
+local function click(x, y)
+    table.insert(steps, function()
+        mouse.coords{x=x, y=y}
+        return true
+    end)
+    table.insert(steps, function()
```

Comment:
You can merge these two into one function, I think. At least I do not see why not.